pkg/processor/reconciler_helper.go

```diff
@@ -153,7 +153,7 @@ func (h *ReconcilerHelper) PruneBalances(
 	currency *types.Currency,
 	index int64,
 ) error {
-	if h.config.Data.PruningDisabled {
```

Comment:
We can fully decouple pruning, so only `h.config.Data.PruningBalanceDisabled` is needed here